Multisig/UI/Transactions/TransactionDetails/TransactionDetailsViewModel.swift

```diff
@@ -86,4 +86,26 @@ class TransactionDetailsViewModel: BasicLoadableViewModel {
             return try App.shared.clientGatewayService.transactionDetails(id: id!)
         }
     }
+
+    func sign(safeAddress: Address) {
+        guard let transferTx = transactionDetails as? TransferTransactionViewModel,
+              let transaction = transferTx.transaction else { return }
```

Comment:
would it be programmer error if  `sign(..)` is called with wrong transactionDetails or the transaction is nil?

Reply:
It would. Would you prefer to crash in guard statement?

Reply:
I would crash only in development builds, i.e. I would add `assertionFailure(...)` - not `preconditionFailure(...)`

Reply:
ok. I will place here `assertionFailure("We have a technical problem. You need to restart the app.")`

Reply:
in the next pr